lib/improver/tests/utilities/test_pad_spatial.py

```diff
@@ -354,6 +355,47 @@ def test_halo_using_mean_smoothing(self):
         self.assertArrayAlmostEqual(padded_cube.data, expected_data)
 
 
+class Test_remove_cube_halo(IrisTest):
+    """Tests for the  remove_cube_halo function"""
+
+    def setUp(self):
+        """Set up a realistic input cube with lots of metadata.  Input cube
+        grid is 1000x1000 km with points spaced 100 km apart."""
+        self.attrs = {'history': '2018-12-10Z: StaGE Decoupler',
+                      'title': 'Temperature on UK 2 km Standard Grid',
+                      'source': 'Met Office Unified Model'}
+        self.cube = set_up_variable_cube(
+            np.ones((3, 11, 11), dtype=np.float32), spatial_grid='equalarea',
+            standard_grid_metadata='uk_det', attributes=self.attrs)
+
+        self.cube_1d = set_up_variable_cube(
+            np.ones((1, 11, 11), dtype=np.float32), spatial_grid='equalarea',
+            standard_grid_metadata='uk_det', attributes=self.attrs)
+        self.grid_spacing = 100000
+
+    def test_basic(self):
+        """Test function returns a cube with expected attributes and shape"""
+        halo_size_km = 162.
+        result = remove_cube_halo(self.cube, 1000.*halo_size_km)
+        self.assertIsInstance(result, iris.cube.Cube)
+        self.assertEqual(result.attributes['history'],
+                         self.attrs['history'])
+        self.assertEqual(result.attributes['title'], self.attrs['title'])
+        self.assertEqual(result.attributes['source'], self.attrs['source'])
```

Comment:
You can just use `self.assertDictEqual` I think (unless it's a limited attributes dict or something awkward like that).